src/sys.rs

```diff
@@ -0,0 +1,248 @@
+
+    /// Check if the current platform is Windows.
+    ///
+    /// This can be set to true in wasm32-unknown-unknown targets that
+    /// are running on Windows systems.
+    fn is_windows(&self) -> bool;
+    /// Gets the current working directory.
+    fn current_dir(&self) -> io::Result<PathBuf>;
+    /// Gets the home directory of the current user.
+    fn home_dir(&self) -> Option<PathBuf>;
+    /// Splits a platform-specific PATH variable into a list of paths.
+    fn env_split_paths(&self, paths: &OsStr) -> Vec<PathBuf>;
+    /// Gets the value of an environment variable.
+    fn env_var_os(&self, name: &OsStr) -> Option<OsString>;
+    fn env_var(&self, key: &OsStr) -> Result<String, VarError> {
+        match self.env_var_os(key) {
+            Some(val) => val.into_string().map_err(VarError::NotUnicode),
+            None => Err(VarError::NotPresent),
+        }
+    }
+    /// Gets and parses the PATHEXT environment variable on Windows.
+    ///
+    /// Override this to enable caching the parsed PATHEXT.
+    ///
+    /// Note: This will only be called when `is_windows()` returns `true`
+    /// and isn't conditionally compiled with `#[cfg(windows)]` so that it
+    /// can work in Wasm.
+    fn env_windows_path_ext(&self) -> Cow<'static, [String]> {
+        Cow::Owned(
+            self.env_var(OsStr::new("PATHEXT"))
+                .map(|pathext| parse_path_ext(&pathext))
+                // PATHEXT not being set or not being a proper Unicode string is exceedingly
+                // improbable and would probably break Windows badly. Still, don't crash:
+                .unwrap_or_default(),
+        )
+    }
+    /// Gets the metadata of the provided path, following symlinks.
+    fn metadata(&self, path: &Path) -> io::Result<Self::Metadata>;
+    /// Gets the metadata of the provided path, not following symlinks.
+    fn symlink_metadata(&self, path: &Path) -> io::Result<Self::Metadata>;
+    /// Reads the directory entries of the provided path.
+    fn read_dir(
+        &self,
+        path: &Path,
+    ) -> io::Result<Box<dyn Iterator<Item = io::Result<Self::ReadDirEntry>>>>;
+    /// Checks if the provided path is a valid executable.
+    fn is_valid_executable(&self, path: &Path) -> io::Result<bool>;
+}
+
+impl SysReadDirEntry for std::fs::DirEntry {
+    fn file_name(&self) -> OsString {
+        self.file_name()
+    }
+
+    fn path(&self) -> PathBuf {
+        self.path()
+    }
+}
+
+impl SysMetadata for std::fs::Metadata {
+    fn is_symlink(&self) -> bool {
+        self.file_type().is_symlink()
+    }
+
+    fn is_file(&self) -> bool {
+        self.file_type().is_file()
+    }
+}
+
+#[cfg(feature = "real-sys")]
+#[derive(Default, Clone)]
+pub struct RealSys;
+
+#[cfg(feature = "real-sys")]
+impl RealSys {
+    #[inline]
+    pub(crate) fn canonicalize(&self, path: &Path) -> io::Result<PathBuf> {
+        #[allow(clippy::disallowed_methods)] // ok, sys implementation
+        std::fs::canonicalize(path)
```

Comment:
Ah, maybe these ones? This is done to prevent having to have conditions on `use` statements, which starts to get unwieldy IMO.